str/helper/vcf_pruner.py

```diff
@@ -0,0 +1,155 @@
+
+The output is a pruned VCF file, sharded in the same way as the files in vcf-catalog-dir, output to a GCS bucket.
+
+Optional to provide multiple samples and create pruned sharded VCFs for each sample.
+
+analysis-runner --access-level test --dataset tob-wgs --description \
+    'VCF pruner' --output-dir 'str/5M_run_combined_vcfs_pruned/v2' \
+    vcf_pruner.py \
+    --vcf-catalog-dir=gs://cpg-tob-wgs-test/str/5M_3M_merge_experiment/3M/CPGXXX \
+    --vcf-file-dir=gs://cpg-tob-wgs-test/str/5M_3M_merge_experiment CPGXXXXX
+"""
+import click
+
+from cpg_utils import to_path
+from cpg_utils.hail_batch import get_batch, output_path
+
+
+def variant_id_collector(catalog_file):
+    """Collects all variant IDs from a sharded VCF file and returns a list of variant IDs."""
+
+    variant_ids = []
+
+    with to_path(catalog_file).open() as f:
+        for line in f:
+            if not line.startswith('#'):
+                # wrangle variant ID from INFO column, assumes ExpansionHunter VCF format
+                row_info = line.split('\t')[7]
+                var_id = (row_info.split(';')[4])[6:]
+
+                variant_ids.append(var_id)
+
+    return variant_ids
+
+
+def pruner(vcf_file_path, cpg_id, chunk_number, variant_id_order):
+    """Prunes a VCF file, removing all variants that are not present in the list of provided variant IDs.
+    Writes output to GCS bucket.
+    """
+    # Initialize variables to store information
+    fileformat_line = ''
+    info_lines = []
+    alt_lines = set()
+    chrom_line = ''
+    gt_lines = {}  # key = variant ID, value = line in VCF containing GT
+    # set structure allows for faster intersection
+    variant_id_set = set(variant_id_order)
+
+    with to_path(vcf_file_path).open() as f:
+        for line in f:
+            # Collect information from the header lines
+            if line.startswith('##fileformat'):
+                fileformat_line = line
+            elif (
+                line.startswith('##INFO')
+                or line.startswith('##FILTER')
+                or line.startswith('##FORMAT')
+            ):
+                info_lines.append(line)
+            elif line.startswith('##ALT'):
+                # Collect ALT lines from all files into a set to remove duplicates
+                alt_lines.add(line)
+            elif line.startswith('#CHROM'):
+                chrom_line = line
+            elif not line.startswith('#'):
+                # Collect calls after #CHROM
+                row_info = line.split('\t')[7]
+                var_id = {(row_info.split(';')[4])[6:]}
+
+                if (
+                    var_id & variant_id_set
+                ):  # if the variant ID is in the set of target variant IDs
+                    var_id = ''.join(
+                        map(str, var_id)
+                    )  # converts var_id from set to string
+                    gt_lines[var_id] = line
```

Comment:
Does each row have a single variant ID, or multiple? The check of 'is a string in a set of strings' is faster than 'does this set of strings intersect with this set'. I pulled an example file and the VARID looks singular
```END=1234;REF=1;RL=12;RU=A;VARID=1-1122-1234-A;REPID=1-1122-1234-A```

```suggestion
                row_info = line.split('\t')[7]
                var_id = row_info.split(';')[4].removeprefix('VARID=')

                # if the variant ID is in the set of target variant IDs
                if var_id in variant_id_set:
                    gt_lines[var_id] = line
```

Reply:
this could be more terse, but would need checking 
```
if var_id := row_info.split(';')[4].removeprefix('VARID=') in variant_id_set:
    gt_lines[var_id] = line
```